File: Baza/DataLayerLib/BuisnessLogicLayer/Services/CommentService.cs
using BuisnessLogicLayer.DAOInterfaces;
using ObjectModel.DTOs;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace BuisnessLogicLayer.Services
{
    public class CommentService
    {
        private CommentData commentDataAccess;
        private CommentService()
        {

        }
        public CommentService(CommentData commentData)
        {
            commentDataAccess = commentData;
        }

        public IEnumerable<CommentDTO> GetAllComments()
        {
            return commentDataAccess.GetAllComments();
        }

        public CommentDTO GetCommentById(int id)
        {
            return commentDataAccess.GetComment(id);
        }
        
        public IEnumerable<CommentDTO> GetCommentsFromNews(int newsId)
        {
            return commentDataAccess.GetCommentsForNews(newsId);
        }
        
        public bool UpdateComment(int id, string content)
        {
            return commentDataAccess.UpdateComment(id, content);
        }
        
        public bool CreateComment(CommentSimpleDTO value)
        {
            return commentDataAccess.CreateComment(value.CreatedBy, value.BelongsToNewsId, value.Content);
        }
        
        public bool DeleteComment(int id)
        {
            return commentDataAccess.DeleteComment(id);
        }
    }
}

File: Android/Android/app/src/main/java/com/newsman/newsman/new_rest/retrofit_services/CommentService.java
package com.newsman.newsman.new_rest.retrofit_services;

import com.newsman.newsman.new_rest.dtos.CommentDTO;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.PUT;
import retrofit2.http.Path;

public interface CommentService {
    @GET("Comment")
    Call<List<CommentDTO>> getAllComments();

    @GET("Comment/{commentId}")
    Call<CommentDTO> getComment(@Path("commentId") int commentId);

    @GET("Comment/FromNews/{newsId}")
    Call<List<CommentDTO>> getCommentsForNews(@Path("newsId") int newsId);

    @POST("Comment/{userId}?content={content}")
    Call<CommentDTO> createComment(@Path("userId") int userId, @Path("content") String content);

    @PUT("Comment")
    Call<CommentDTO> createComment(@Body CommentDTO comment);

    @DELETE("Comment/{commentId}")
    Call<Boolean> deleteComment(@Path("commentId") int commentId);
}
